pkg/manager/manager.go

```diff
@@ -494,6 +494,11 @@ func (o Options) AndFromOrDie(loader config.ControllerManagerConfiguration) Opti
 }
 
 func (o Options) setLeaderElectionConfig(obj v1alpha1.ControllerManagerConfigurationSpec) Options {
+	if obj.LeaderElection == nil {
```

Comment:
Emm... I don't think we can return `o Options` directly if `obj.LeaderElection` is nil. The leaderelection fields in `o Options` should be set default values below.

Reply:
What should those default values be, if `obj.LeaderElection` is not provided? There are no set operations on `o` that don't read from `obj.LeaderElection` in this function.

This code seems to assume that `obj.LeaderElection` will, at the very least, be initialized to a struct value with the defaults set; equivalent to something like
```
if obj.LeaderElection == nil {
    obj.LeaderElection = LeaderElectionConfiguration{}
}
```
However, if one does that, _none_ of the `if` statements below this will evaluate to true, which means the function will return `o` unmodified. This is exactly what my PR accomplishes as well.

Reply:
@FillZpp I pushed a test that panics without this check. If this check is not the way that panic should be avoided, could you please outline what you think should be done instead?

Reply:
Oh, apologies for my misunderstanding. You are right, thanks!